x-pack/plugins/ml/public/application/trained_models/navigation_bar.tsx

```diff
@@ -21,6 +22,8 @@ export const TrainedModelsNavigationBar: FC<{
 }> = ({ selectedTabId }) => {
   const navigateToPath = useNavigateToPath();
 
+  const canViewMlNodes = checkPermission('canViewMlNodes');
```

Comment:
it looks like this capability check should be added to the `nodes_overview` endpoint too

Reply:
Done in bf328c89b954a86419d71a7ac32fd469680df9e1